### PR TITLE
Adding support for net 7.0

### PR DESCRIPTION
**Pull request by JRahnama**
Since net8.0 is in preview, we have added support for net7.0 to be able to have access to some of new APIs.When Net8 is released net7.0 will be replaced with net8.0 for GA release.

### CODE REVIEW

**Comment by roji** [comment]
@JRahnama nice to see this... Out of curiosity, which .NET 7.0 features are you specifically looking at?

**Comment by JRahnama** [comment]
@roji, the goal is to be able to completely eliminate the SqlTypeWorkarounds.cs file for Net 8 and greater versions, but since Net 8 is still in preview stage we decided to go by Net7.0 for preview releases and when net8 GA version is released replace Net7.0 with Net8.

**Comment by Wraith2** [comment]
once this is merged if you change the net8 to net7 in my PR https://github.com/dotnet/SqlClient/pull/1934 then new paths should light up and remove the unneeded workarounds.

File: tools/specs/add-ons/Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyVaultProvider.nuspec [hunk 1]
```diff
@@ -30,6 +30,13 @@ Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyStoreProvider.SqlColumnEncrypti
         <dependency id="System.Text.Encodings.Web" version="6.0.0" />
         <dependency id="Azure.Security.KeyVault.Keys" version="[4.4.0,5.0.0)" />
         <dependency id="Microsoft.Extensions.Caching.Memory" version="6.0.1" />
+      </group>
+       <group targetFramework="net7.0">
```
Comment (David-Engel) [inline tools/specs/add-ons/Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyVaultProvider.nuspec:34]: NIT
```suggestion
      <group targetFramework="net7.0">
```

File: tools/specs/Microsoft.Data.SqlClient.nuspec [hunk 2]
```diff
@@ -97,6 +112,11 @@ When using NuGet 3.x this package requires at least version 3.4.</description>
           <reference file="Microsoft.Data.SqlClient.dll" />
           <reference file="Microsoft.Data.SqlClient.pdb" />
           <reference file="Microsoft.Data.SqlClient.xml" />
+      </group>
+       <group targetFramework="net7.0">
```
Comment (DavoudEshtehari) [inline tools/specs/Microsoft.Data.SqlClient.nuspec:116]:
```suggestion
      <group targetFramework="net7.0">
```

File: src/Microsoft.Data.SqlClient/netcore/src/.editorconfig [hunk 3]
```diff
@@ -10,3 +10,6 @@ csharp_style_implicit_object_creation_when_type_is_apparent = false
 
 # IDE0063: Use simple 'using' statement
 csharp_prefer_simple_using_statement = false
+
+# SYSLIB0039: Type or member is obsolete
+dotnet_diagnostic.SYSLIB0039.severity = none
```
Comment (DavoudEshtehari) [inline src/Microsoft.Data.SqlClient/netcore/src/.editorconfig:15]: I'd rather keep the report in the error list and unblock the build.
```suggestion
dotnet_diagnostic.SYSLIB0039.severity = suggestion
```

**Review by DavoudEshtehari — APPROVED** [review]

**Review by Kaur-Parminder — APPROVED** [review]

**Pull request merged**